src/turing_machine/machine.py

```diff
@@ -17,27 +21,35 @@ def get_value(self):
     def set_value(self, val):
         self.tape[self.head] = val
 
-    def __str__(self):
-        return ''.join(self.tape)
+    def clone(self):
+        return Tape(self.tape[:])
 
+    def __str__(self):
+        tape_str = ", ".join(self.tape)
+        tape_str_with_head = tape_str[:self.head*3] + f"[{self.tape[self.head]}]" + tape_str[self.head*3 + 1:]
```

Comment:
Niekoniecznie każdy symbol na taśmie musi mieć długość akurat jednego znaku. W takim wypadku ta metoda nie zadziała.

Zrób to w ten sposób: `tape_str_with_head = ", ".join(self.tape[:self.head] + f"[{self.tape[self.head]}], " + ", ".join(self.tape[self.head + 1:]`